cypress/e2e/expenses/Add.cy.js

```diff
@@ -0,0 +1,105 @@
+import '@testing-library/cypress/add-commands';
+
+import {
+  getUKFormattedEuros,
+  getUKFormattedDate,
+  getCurrentMonth,
+} from '../../../src/utils/helpers';
+
+describe('ADD EXPENSE - NOT CURRENT MONTH', () => {
+  before(() => {
+    cy.intercept('GET', '**/items*', []).as('getExpenses');
+    cy.visit('/');
+    cy.wait('@getExpenses');
+  });
+
+  beforeEach(() => {
+    cy.intercept('POST', '**/items*', { fixture: 'expense-post' }).as(
+      'postExpense'
+    );
+  });
+
+  it('adds the expense', () => {
+    cy.addExpenseFromFixture('expense-raw');
+    cy.wait('@postExpense');
+  });
+
+  it('selects expense month', () => {
+    cy.fixture('expense-raw').then(expense => {
+      const expenseMonth = getUKFormattedDate(expense.date, {
+        year: 'numeric',
+        month: 'long',
+      });
+      cy.get(`[id="New expense added to ${expenseMonth}"]`).within(() => {
+        cy.get('[data-cy="okButton"]').click();
+      });
+      cy.findByRole('combobox', { name: /select a month/i }).should(
+        'have.value',
+        expenseMonth
+      );
+    });
+  });
+
+  it('it closes the alert', () => {
```

Comment:
i am not familiar with cypress in particular. i only know from RSpec (which inspired cypress), an `it` should be **self contained** and different `it`s may be run in any order (or in parallel).

i think this is true for our test cases, is it?

Reply:
@motine: I took a quick look to RSpec and it looks almost identical to Cypress. They use `describe()` with one or many `it()` blocks within. Yes, looks exactly the same. It() blocks run consecutively and hooks (before, beforeEach, after, afterEach) run in parallel yes